packages/client/hmi-client/src/workflow/ops/model-config/tera-model-config-editor.vue

```diff
@@ -0,0 +1,221 @@
+					<li v-for="[init, vals] in initials" :key="init">
+						<Button
+							:label="init"
+							class="asset-button"
+							plain
+							text
+							size="large"
+							@click="getClickedField('initial', vals[0])"
+						/>
+					</li>
+				</ul>
+			</div>
+			<div>
+				<h4>Parameters</h4>
+				<Accordion v-if="stratifiedModelType && !_.isEmpty(parameters)" :multiple="true">
+					<AccordionTab v-for="[param, vals] in parameters" :key="param">
+						<template #header>
+							{{ param }}
+						</template>
+						<ul>
+							<li v-for="val in vals" :key="val">
+								<Button
+									:label="val"
+									class="asset-button"
+									plain
+									text
+									size="small"
+									@click="getClickedField('parameter', val)"
+								/>
+							</li>
+						</ul>
+					</AccordionTab>
+				</Accordion>
+				<ul v-else-if="!stratifiedModelType && !_.isEmpty(parameters)">
+					<li v-for="[param, vals] in parameters" :key="param">
+						<Button
+							:label="param"
+							class="asset-button"
+							plain
+							text
+							size="large"
+							@click="getClickedField('parameter', vals[0])"
+						/>
+					</li>
+				</ul>
+			</div>
+		</section>
+		<section>
+			<div class="content-container">
+				<div v-if="fieldType === 'parameter'" class="form-section">
+					<h4>
+						{{
+							currentParam?.name ? `${currentParam.name} | ${currentParam.id}` : currentParam?.id
+						}}
+					</h4>
+					<div v-if="currentParam?.description">
+						<h3>Description</h3>
+						<p>{{ currentParam?.description }}</p>
+					</div>
+					<div>
+						<h3>Value</h3>
+						<InputNumber
+							class="p-inputtext-sm"
+							inputId="numericInput"
+							mode="decimal"
+							v-model="paramValue"
+							@update:model-value="updateParameter(paramValue)"
+						/>
+					</div>
+				</div>
+				<div v-else-if="fieldType === 'initial'" class="form-section">
+					<h4>{{ currentInitial?.target }}</h4>
+					<div>
+						<h3>Expression</h3>
+						<InputText
+							class="p-inputtext-sm"
+							v-model="initialExpression"
+							@update:model-value="updateInitial(initialExpression)"
+						/>
+					</div>
+				</div>
+				<h4 v-else>Select a field to edit</h4>
+			</div>
+		</section>
+	</div>
+</template>
+
+<script setup lang="ts">
+import _ from 'lodash';
+import { computed, ref, watch } from 'vue';
+import Accordion from 'primevue/accordion';
+import AccordionTab from 'primevue/accordiontab';
+import Button from 'primevue/button';
+import InputNumber from 'primevue/inputnumber';
+import InputText from 'primevue/inputtext';
+import { Model, Initial, ModelParameter } from '@/types/Types';
+import {
+	getUnstratifiedParameters,
+	getUnstratifiedInitials
+} from '@/model-representation/petrinet/mira-petri';
+import { getStratificationType } from '@/model-representation/petrinet/petrinet-service';
+
+const props = defineProps<{
+	model: Model;
+	initials?: Initial[];
+	parameters?: ModelParameter[];
+}>();
```

Comment:
Would it be easier to make **initials** and **parameters** required? As in, even if a model has no parameters it would be an empty array instead of null.

Reply:
done